src/plugins/navigation_embeddable/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -86,6 +91,52 @@ export const DashboardLinkComponent = ({
     };
   }, [error, dashboardTitle, dashboardDescription]);
 
+  /**
+   * Dashboard-to-dashboard navigation
+   */
+  const { loading: loadingOnClickProps, value: onClickProps } = useAsync(async () => {
+    /** If the link points to the current dashboard, then there should be no `onClick` or `href` prop */
+    if (link.destination === parentDashboardId) return;
+
+    const linkOptions = {
+      ...DEFAULT_DASHBOARD_DRILLDOWN_OPTIONS,
+      ...link.options,
+    } as DashboardDrilldownOptions;
+
+    const locator = await getDashboardLocator({
+      link: { ...link, options: linkOptions },
+      navEmbeddable,
+    });
+    if (!locator) return;
+
+    const href = getDashboardHref(locator);
+    return {
+      href,
```

Comment:
`href` is always necessary so that right clicking the link gives the `"Open in new <tab/window>"` option.